chaincode/src/main/java/de/upb/cs/uc4/chaincode/MatriculationDataChaincode.java

```diff
@@ -32,100 +29,123 @@ public void initLedger(final Context ctx) {
 
     /**
      * Adds MatriculationData to the ledger.
-     * @param ctx
+     * @param ctx transaction context providing access to ChaincodeStub etc.
      * @param jsonMatriculationData json-representation of a MatriculationData to be added
      * @return Empty string on success, serialized error on failure
      */
     @Transaction()
     public String addMatriculationData(final Context ctx, final String jsonMatriculationData) {
-        _logger.info("immatriculateMatriculationData");
 
         ChaincodeStub stub = ctx.getStub();
         MatriculationData matriculationData;
 
         try {
-            matriculationData = gson.fromJson(jsonMatriculationData, MatriculationData.class);
+            matriculationData = GSON.fromJson(jsonMatriculationData, MatriculationData.class);
         } catch(Exception e) {
-            return gson.toJson(new GenericError()
+            return GSON.toJson(new GenericError()
                     .type("hl: unprocessable entity")
                     .title("The given parameter does not conform to the specified format."));
         }
 
         String result = stub.getStringState(matriculationData.getMatriculationId());
         if (result != null && !result.equals("")) {
-            return gson.toJson(new GenericError()
+            return GSON.toJson(new GenericError()
                     .type("hl: conflict")
                     .title("There is already a MatriculationData for the given matriculationId."));
         }
 
         ArrayList<InvalidParameter> invalidParams = getErrorForMatriculationData(matriculationData);
 
-        if(!invalidParams.isEmpty()){
-            return gson.toJson(new DetailedError()
+        if (!invalidParams.isEmpty()) {
+            return GSON.toJson(new DetailedError()
                     .type("hl: unprocessable field")
                     .title("The following fields in the given parameters do not conform to the specified format.")
                     .invalidParams(invalidParams));
         }
 
-        stub.putStringState(matriculationData.getMatriculationId(),gson.toJson(matriculationData));
+        stub.putStringState(matriculationData.getMatriculationId(), GSON.toJson(matriculationData));
         return "";
     }
 
+    /**
+     * Updates MatriculationData on the ledger.
+     * @param ctx transaction context providing access to ChaincodeStub etc.
+     * @param jsonMatriculationData json-representation of the updated MatriculationData
```

Comment:
is "updated" the right word here? Maybe confusing, because this is the param for the "new" MatriculationData.

Reply:
How about
```suggestion
     * @param jsonMatriculationData json-representation of the new MatriculationData to replace the old with
```